### PR TITLE
Add netcore support to Azure.Batch

### PR DESCRIPTION
**Pull request by matthchr**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by matthchr** [comment]
## Reviewers please note:
1. **Do not merge this pull request just yet (feel free to review it though) -- I am just making sure that the full build works correctly right now. I will update this comment once we are 100% ready to merge**
2. Note the "Avoid packaging folders containing "tests" commit, where I changed the build a bit. This should be reviewed by you SDK folks

**Comment by matthchr** [comment]
Travis is failing with this error:
```
Errors in /home/travis/build/Azure/azure-sdk-for-net/src/Batch/Client/Tests/ObjectModel/Azure.Batch.Unit.Tests/project.json
    Unable to resolve 'Azure.Batch (>= 6.0.1)' for '.NETCoreApp,Version=v1.0'.
```

I am not sure why Travis would fail with this while Jenkins succeeded...

**Comment by matthchr** [comment]
Fixed Travis -- it was a capitalization issue in the `global.json` that was causing project-based package restores to fail.

@cormacpayne - Who should I bug to review this?

**Comment by cormacpayne** [comment]
@markcowl can you take a look?

**Comment by markcowl** [comment]
@matthchr It looks like there's no package.json in the diff.  How are you building the NetCore nuget package without it?

**Comment by matthchr** [comment]
@markcowl `package.json`? I don't see a `package.json` anywhere in the source tree of this repo (i.e. in ResourceManagement). Do you mean `project.json`? If so, we have `project.json` added at src\Batch\Client\Src\Azure.Batch - the project.json is what we're using to build via netcore and produce a package.

I did a local build and a package was correctly produced containing net45 and netstandard1.5 assemblies in the binaries folder at the root of the repo.

File: src/Batch/Client/Src/Azure.Batch/project.json
Comment (markcowl) [inline src/Batch/Client/Src/Azure.Batch/project.json]: To be VS2017 compatible, you need to use ```NetStandard.Library: 1.6.1```  this makes the dependencies on System.Net.Http, S.R.S.Primitives, and System.Threading.Tasks unnecessary

**Comment by markcowl** [comment]
@matthchr Yes, my bad, added one comment.  Building the packages to verify, things look good other than this.

**Review by markcowl — CHANGES_REQUESTED** [review]

**Comment by matthchr** [comment]
@markcowl I fixed the the `NetStandard.Library` issue you pointed out.

**Review by markcowl — APPROVED** [review]

**Pull request merged**
